### ISSUE TITLE
Add dispose functionality for geometries and textures

### ISSUE THREAD

**Issue by shlomnissan** [chanzuckerberg/idetik#41]
Currently, we allocate names and buffers for geometries and textures using the WebGL renderer but never deallocate them. This is especially crucial for textures, as we might need to free up GPU memory that's no longer in use before uploading new texture data.

**Comment by andy-sweet** [comment]
@shlomnissan : are you still actively working on this or planning to in the near future? If not, can you unassign? I think this is still relevant and worth doing, but if that's not the case, please close this issue.

**Comment by shlomnissan** [comment]
Yep, still relevant. Thanks for going through the issues :) Let's keep this one open / assigned to me
